speechbrain/dataio/dataio.py

```diff
@@ -66,32 +71,34 @@ def load_data_json(json_path, replacements={}):
     """
     with open(json_path, "r") as f:
         out_json = json.load(f)
-    _recursive_format(out_json, replacements)
+    _recursive_format(out_json, replacements, target_columns)
     return out_json
 
 
-def _recursive_format(data, replacements):
+def _recursive_format(data, replacements, target_columns=None):
     # Data: dict or list, replacements : dict
     # Replaces string keys in replacements by their values
     # at all levels of data (in str values)
     # Works in-place.
     if isinstance(data, dict):
         for key, item in data.items():
-            if isinstance(item, dict) or isinstance(item, list):
-                _recursive_format(item, replacements)
-            elif isinstance(item, str):
-                data[key] = item.format_map(replacements)
-            # If not dict, list or str, do nothing
+            if target_columns is None or key in target_columns:
+                if isinstance(item, dict) or isinstance(item, list):
+                    _recursive_format(item, replacements)
+                elif isinstance(item, str):
+                    data[key] = item.format_map(replacements)
+                # If not dict, list or str, do nothing
     if isinstance(data, list):
         for i, item in enumerate(data):
-            if isinstance(item, dict) or isinstance(item, list):
-                _recursive_format(item, replacements)
-            elif isinstance(item, str):
-                data[i] = item.format_map(replacements)
-            # If not dict, list or str, do nothing
+            if target_columns is None or i in target_columns:
```

Comment:
Does this functionality even make sense to have for list indices? This seems like unintended behavior and ambiguous especially with nested structures.